### PR TITLE
Javadoc for using Realm with SubscribedOn() 

### PR DESCRIPTION
**Pull request by stk1m1**
It is recommended to use Realm.where().find*Async() instead of SubscribedOn() to avoid Realm being accessed from different threads.

This PR addresses issue #2607.
@realm/java

### CODE REVIEW

**Comment by kneth** [comment]
:+1:

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 1]
```diff
@@ -162,6 +162,9 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
      * }
      * </pre>
      *
+     * <p>Note that when the Realm is accessed from threads other than where it was created, {@link IllegalStateException}
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: This issue is about RealmObject not Realm, so this should ideally be moved to RealmResults/RealmObject instead.

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 2]
```diff
@@ -162,6 +162,9 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
...
+     * will be thrown. Please avoid using {@code subscribeOn()}, and use {@code Realm.where().*Async()} methods instead.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Actually it is the combination of observeOn/subscribeOn that is problematic, since that will cause the thread changes, so I would probably rephrase this to:

"Care should  be taken when using different schedulers with {@code subscribeOn()} and {@code observeOn}. Consider using {@code Realm.where().find*Async()} instead."

**Comment by stk1m1** [comment]
The review points are addressed.

File: realm/realm-library/src/main/java/io/realm/RealmObject.java [hunk 3]
```diff
@@ -327,6 +327,10 @@ public final void removeChangeListeners() {
      * }
      * </pre>
      *
+     * <p>Note that when the {@link Realm} is accessed from threads other than where it was created,
+     * {@link IllegalStateException} will be thrown. Please avoid using {@code subscribeOn()}, and use
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmObject.java]: Imo this still doesn't capture it, pr. my previous review. It is the combination of subscribeOn/observeOn that is problematic.

Also, DynamicRealmObject also have a asObservable method :)

**Comment by stk1m1** [comment]
I don't think `DynamicRealmObject`has overridden `asObservable()` method. Review points are addressed.

**Comment by cmelchior** [comment]
👍
